FIAT/macro.py

```diff
@@ -0,0 +1,186 @@
+        super(SplitSimplicialComplex, self).__init__(ref_el.shape, vertices, topology)
+
+    def split_topology(self, ref_el):
+        raise NotImplementedError
+
+    def get_child_to_parent(self):
+        bary = xy_to_bary(numpy.asarray(self.parent.get_vertices()),
+                          numpy.asarray(self.get_vertices()))
+        top = self.get_topology()
+        parent_inv_top = invert_cell_topology(self.parent.get_topology())
+        child_to_parent = {}
+        for dim in top:
+            child_to_parent[dim] = {}
+            for entity in top[dim]:
+                facet_ids = top[dim][entity]
+                facet_coords = bary[list(facet_ids), :]
+                parent_verts = facet_support(facet_coords)
+                parent_dim = len(parent_verts) - 1
+                parent_entity = parent_inv_top[parent_dim][parent_verts]
+                child_to_parent[dim][entity] = (parent_dim, parent_entity)
+        return child_to_parent
+
+    def get_cell_connectivity(self):
+        """Connectitivity from cell in a complex to globally number and
+        respects the entity numbering on the reference cell.
+        """
+        sd = self.get_spatial_dimension()
+        top = self.get_topology()
+        inv_top = invert_cell_topology(top)
+        parent_top = self.parent.get_topology()
+        connectivity = {cell: {dim: [] for dim in top} for cell in top[sd]}
+        for cell in top[sd]:
+            cell_verts = top[sd][cell]
+            for dim in top:
+                for entity in parent_top[dim]:
+                    ref_verts = parent_top[dim][entity]
+                    global_verts = tuple(cell_verts[v] for v in ref_verts)
+                    connectivity[cell][dim].append(inv_top[dim][global_verts])
+        return connectivity
+
+    def construct_subelement(self, dimension):
+        """Constructs the reference element of a cell subentity
+        specified by subelement dimension.
+
+        :arg dimension: subentity dimension (integer)
+        """
+        return self.parent.construct_subelement(dimension)
+
+
+class AlfeldSplit(SplitSimplicialComplex):
+
+    def split_topology(self, ref_el, splits=1, variant=None):
+        assert splits == 1
+        sd = ref_el.get_spatial_dimension()
+        top = ref_el.get_topology()
+        new_topology = copy.deepcopy(top)
+        new_topology[sd] = {}
+
+        barycenter = ref_el.make_points(sd, 0, sd+1)
+        old_verts = ref_el.get_vertices()
+        new_verts = old_verts + tuple(barycenter)
+        new_vert_id = len(old_verts)
+
+        new_topology[0][new_vert_id] = (new_vert_id,)
+        for dim in range(1, sd + 1):
+            offset = len(new_topology[dim])
+            for entity, ids in top[dim-1].items():
+                new_topology[dim][offset+entity] = ids + (new_vert_id,)
+        return new_verts, new_topology
+
+
+class IsoSplit(SplitSimplicialComplex):
+
+    def split_topology(self, ref_el, splits=1, variant=None):
```

Comment:
`splits` should be renamed -- the idea is we support the kind of split with a k:1 ratio (at least topologically).  Maybe default value of `2` would give the splitting used for P2-P1iso.
And what is `variant`?  The kind of Lagrange points (e.g. equispaced, gll)?  Maybe change to `point_type` or `point_variant`?